relay-server/src/utils/rate_limits.rs

```diff
@@ -93,7 +93,7 @@ fn infer_event_category(item: &Item) -> Option<DataCategory> {
         ItemType::Event => Some(DataCategory::Error),
         ItemType::Transaction => Some(DataCategory::Transaction),
         ItemType::Security | ItemType::RawSecurity => Some(DataCategory::Security),
-        ItemType::Nel => None,
+        ItemType::Nel => Some(DataCategory::Error),
```

Comment:
Good catch. Should we add a test that shows that `Nel` reports are removed when there is an error quota?